docs/tendermint-core/consensus/proposer-based-timestamps.md

```diff
@@ -0,0 +1,89 @@
+---
+order: 3
+---
+
+# Proposer-Based Timestamps
+
+ This document provides an overview of the Proposer-Based Timestamp algorithm 
+ added to Tendermint in the v0.36 release. This document outlines the core functionality
+ as well as the parameters and constraints of the this algorithm.
+
+# Algorithm Overview 
+
+The proposer-based timestamps algorithm defines a way for a Tendermint blockchain
+to create block timestamps that are within a reasonable bound of the clocks of the
+validators on the network.
+
+## Algorithm Parameters
+
+The functionality of the Proposer-Based Timestamps algorithm is governed by two
+parameters within Tendermint. These two parameters are [consensus parameters](https://github.com/tendermint/tendermint/blob/c42c6d06d2986791ad1cb8742824818edd9b6323/spec/abci/apps.md#L291), 
+meaning they are configured by the ABCI application and are expected to be the
+same across all nodes on the network.
+
+### `Precision`
+
+The `Precision` parameter configures the acceptable upper-bound of clock drift among
+all of the nodes on a Tendermint network. Any two nodes on a Tendermint network
+are expected to have clocks that differ by at most `Precision` seconds? Ms? at
+any given instant.
```

Comment:
We should pick a unit here. I imagine milliseconds is sufficiently fine-grained.